src/main/java/build/buildfarm/common/redis/RedisClient.java

```diff
@@ -109,8 +144,47 @@ public <T> T blockingCall(JedisInterruptibleContext<T> withJedis)
     return result;
   }
 
-  @SuppressWarnings("ConstantConditions")
   public <T> T call(JedisContext<T> withJedis) throws IOException {
+    return callImpl(withJedis);
+  }
+
+  private <T> T callImpl(JedisContext<T> withJedis) throws IOException {
+    // Typical configuration that does not handle exceptions
+    // or try to restablish the jedis client on failures.
+    if (!restablishClientOnFailures) {
+      return defaultCall(withJedis);
+    }
+
+    // Alternatively,
+    // Capture all redis problems at the client level.
+    // Try to re-establish the client and log all issues.
+    // This will block the overall thread until redis can be connected to.
+    // It may be a useful strategy for gaining stability on a poorly performing network,
+    // or a redis cluster that goes down.
+    while (true) {
```

Comment:
I would prefer configuration to be a number of a reconnects. If it's 0 or null then we don't retry, otherwise we retry up to that number of times. Bonus if there is some backoff here where we don't spam retries continuously.

Reply:
Agree.  Switched to retry amount + duration between retries.  We have a `Retrier.java` class in this repo, but its too specific to grpc so I didn't use it.  Might be nice to use something like https://resilience4j.readme.io/docs/retry in the future.